### DISCUSSION TITLE
part-ii/2001-3/

### DISCUSSION THREAD

**Discussion by tripos-comments** [tripos-education/maths-tripos-questions#142] [Question comments]
Part II 2001 Graph Theory: A1.8 

https://questions.tripos.org/part-ii/2001-3/

**Comment by walrus1729** [comment]
<details>
<summary>Solution to (i)</summary>

Define a sequence of vertices $v_1 v_2 v_3 \dots$ chosen such that $v_1$ is any vertex, and $v_{i+1}$ is a neighbour of $v_i$ that has not previously appeared in the sequence. We continue until this process terminates. We will definitely have at least $\delta + 1$ vertices in the sequence because for $i \leq \delta$, there are more vertices in $N(v_i)$ than appeared in the sequence before $v_i$, so we can always add another vertex. If this process terminates at $v_i$, all of the neighbours of $v_i$ appeared previously in the sequence. But $d(v_i) \leq \delta$, so there exists $j$ with $i - j \geq \delta$ such that $v_i \sim v_j$. Then $v_j v_{j+1} \dots v_{i-1} v_i$ is a cycle of length at least $\delta + 1$.

</details>

<details>
<summary>Solution to (ii)</summary>

The first part (Cayley's formula) was presumably bookwork at the time and there are lots of proofs online, but I'll give my favourite anyway.

Let the number of spanning trees in $K_N$ be $\tau(N)$ and count the number of directed spanning trees of $K_N$ with a root and some numbering of the edges $\{1, \dots, N-1\}$ in two ways. One way to count this is to take the number of spanning trees and multiply by the $N$ choices of root and the $(N-1)!$ possible numberings of edges.

Another way is to add edges one at a time. There are $N$ possible endpoints for each edge, and the $k$th edge has $(N-k)$ possible start points, since we must start at the root of a component and cannot start at the same component we end at. This gives the formula

$$\prod_{i=1}^{N-1} N (N - k) = N^{N-1} (N - 1)!$$

Equating the two gives

$$\tau(N) \cdot N \cdot (N - 1)! = N^{N-1} \cdot (N - 1)!$$

so $\tau(N) = N^{N-2}$ as required.

To count the number of trees with $v_1$ having degree 1, we note that every such tree is uniquely determined by a choice of spanning tree in the remaining $N-1$ vertices and a choice of vertex to connect $v_1$ to. There are $(N-1)^{N-3}$ such trees by the previous part, and $N-1$ choices of vertex, so we get a total of $(N-1)^{N-2}$ trees.

Due to the symmetry of $K_N$, we solve this part by diving the number of spanning trees in which $v_1$ has degree 1 by the total number of spanning trees, and multiplying by $N$ since any vertex can have degree 1. We get

$$\frac{N(N-1)^{N-2}}{N^{N-2}} = (N - 1)(1 - \frac{1}{N}^{N - 2} \approx N(1 - \frac{1}{N})^N = Ne^{-1}$$

as required.

For the final part, simply note that a tree on $N$ vertices always has $N - 1$ edges, so the average degree is given by $\frac{2e(G)}{N} = \frac{2(N-1)}{N} = 2 - \frac{2}{N}$.


</details>
